system/security/plugins/jwtSecurity/jwtSecurity.cpp

```diff
@@ -0,0 +1,1149 @@
+            {
+                if (x->claim == "AllowFileScopeAccess")
+                    userInfo->addFileScopePerm(x->scopePattern, SecAccess_Access, defaultFileAccess);
+                else if (x->claim == "AllowFileScopeRead")
+                    userInfo->addFileScopePerm(x->scopePattern, SecAccess_Read, defaultFileAccess);
+                else if (x->claim == "AllowFileScopeWrite")
+                    userInfo->addFileScopePerm(x->scopePattern, SecAccess_Write, defaultFileAccess);
+                else if (x->claim == "AllowFileScopeFull")
+                    userInfo->addFileScopePerm(x->scopePattern, SecAccess_Full, defaultFileAccess);
+                else if (x->claim == "DenyFileScopeAccess")
+                    userInfo->addFileScopePerm(x->scopePattern, SecAccess_Access, defaultFileAccess, true);
+                else if (x->claim == "DenyFileScopeRead")
+                    userInfo->addFileScopePerm(x->scopePattern, SecAccess_Read, defaultFileAccess, true);
+                else if (x->claim == "DenyFileScopeWrite")
+                    userInfo->addFileScopePerm(x->scopePattern, SecAccess_Write, defaultFileAccess, true);
+                else if (x->claim == "DenyFileScopeFull")
+                    userInfo->addFileScopePerm(x->scopePattern, SecAccess_Full, defaultFileAccess, true);
+            }
+
+            gUserCache.set(subject, userInfo);
+
+            verificationResult = true;
+        }
+        catch (const std::runtime_error& e)
+        {
+            // Most jwt-cpp exceptions are subclasses of std::runtime_error
+            PROGLOG("CJwtSecurityManager::verifyToken error: %s", e.what());
+        }
+        catch (const std::logic_error& e)
+        {
+            PROGLOG("CJwtSecurityManager::logic_error: %s", e.what());
+        }
+        catch (const std::bad_cast& e)
+        {
+            PROGLOG("CJwtSecurityManager::bad_cast error: %s", e.what());
+        }
+        catch (...)
+        {
+            PROGLOG("CJwtSecurityManager::verifyToken unknown error");
+            throw;
+        }
+
+        return verificationResult;
+    }
+
+    /**
+     * Load the contents of the key if necessary.  Contents loaded when
+     * at least KEY_FILE_CHECK_INTERVAL have passed since last check time
+     *
+     * Function will throw an exception on error.
+     *
+     * @return  true
+     */
+    bool ensureKeyLoaded()
+    {
+        if ((keyContentsTimer.elapsedMs() > KEY_FILE_CHECK_INTERVAL_MS) || keyContents.empty())
+        {
+            try
+            {
+                StringBuffer    rawKeyContents;
+
+                getSecretValue(rawKeyContents, "system", secretsPath.c_str(), "key", true);
+
+                if (rawKeyContents.isEmpty())
+                    throw makeStringExceptionV(-1, "CJwtSecurityManager: Key at path '%s' is empty", secretsPath.c_str());
+
+                keyContents = trailingStringTrim(trailingStringTrim(rawKeyContents.str()));
```

Comment:
trivial: duplicate call to trailingStringTrim()

Reply:
One of those should have been leadingStringTrim(); changed.